### ISSUE TITLE
When using scout together with datadog, app can't start up

### ISSUE THREAD

**Issue by w32-blaster** [scoutapp/scout_apm_ruby#309]
When we use scout_apm and ddtrace (datadog apm) as well we can't start up the application anymore. This was working before (version > 2.2). 
Here is the trace we get in rails:
"SystemStackError: stack level too deep"

**Comment by dlanderson** [comment]
Apologies @w32-blaster - do you happen to have the versions of ddtrace and scout_apm you're using? If you also have a backtrace go ahead and place it here, too.

**Comment by w32-blaster** [comment]
Sure. 
Versions we use:
scout_apm: latest released
ddtrace: 0.26.0

Here is the full stack trace:
```
SystemStackError: stack level too deep
  from active_support/core_ext/time/calculations.rb:285:in `minus_with_coercion'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from ddtrace/contrib/rails/core_extensions.rb:80:in `render_template_with_datadog'
  from scout_apm/instruments/action_view.rb:128:in `render_template'
  from action_view/renderer/template_renderer.rb:16:in `render'
  from ddtrace/contrib/rails/core_extensions.rb:42:in `block (2 levels) in render_with_datadog'
  from ddtrace/contrib/rails/core_extensions.rb:93:in `with_datadog_span'
  from ddtrace/contrib/rails/core_extensions.rb:42:in `block in render_with_datadog'
  from ddtrace/tracer.rb:303:in `trace'
  from ddtrace/contrib/rails/core_extensions.rb:38:in `render_with_datadog'
  from action_view/renderer/renderer.rb:44:in `render_template'
  from action_view/renderer/renderer.rb:25:in `render'
  from action_view/rendering.rb:103:in `_render_template'
  from action_controller/metal/streaming.rb:219:in `_render_template'
  from action_view/rendering.rb:84:in `render_to_body'
  from action_controller/metal/rendering.rb:52:in `render_to_body'
  from action_controller/metal/renderers.rb:142:in `render_to_body'
  from abstract_controller/rendering.rb:25:in `render'
  from action_controller/metal/rendering.rb:36:in `render'
  from action_controller/metal/instrumentation.rb:46:in `block (2 levels) in render'
  from active_support/core_ext/benchmark.rb:14:in `block in ms'
  from benchmark.rb:308:in `realtime'
  from active_support/core_ext/benchmark.rb:14:in `ms'
  from action_controller/metal/instrumentation.rb:46:in `block in render'
  from action_controller/metal/instrumentation.rb:87:in `cleanup_view_runtime'
  from active_record/railties/controller_runtime.rb:31:in `cleanup_view_runtime'
  from action_controller/metal/instrumentation.rb:45:in `render'
  from health_check/health_check_controller.rb:50:in `block (2 levels) in send_response'
  from action_controller/metal/mime_responds.rb:203:in `respond_to'
  from health_check/health_check_controller.rb:46:in `send_response'
  from health_check/health_check_controller.rb:28:in `index'
  from action_controller/metal/basic_implicit_render.rb:6:in `send_action'
  from abstract_controller/base.rb:194:in `process_action'
  from scout_apm/instruments/action_controller_rails_3_rails4.rb:73:in `process_action'
  from ddtrace/contrib/rails/action_controller_patch.rb:49:in `process_action'
  from action_controller/metal/rendering.rb:30:in `process_action'
  from abstract_controller/callbacks.rb:42:in `block in process_action'
  from active_support/callbacks.rb:109:in `block in run_callbacks'
  from raven/integrations/rails/controller_transaction.rb:7:in `block in included'
  from active_support/callbacks.rb:118:in `instance_exec'
  from active_support/callbacks.rb:118:in `block in run_callbacks'
  from active_support/callbacks.rb:136:in `run_callbacks'
  from abstract_controller/callbacks.rb:41:in `process_action'
  from action_controller/metal/rescue.rb:22:in `process_action'
  from action_controller/metal/instrumentation.rb:34:in `block in process_action'
  from active_support/notifications.rb:168:in `block in instrument'
  from active_support/notifications/instrumenter.rb:23:in `instrument'
  from active_support/notifications.rb:168:in `instrument'
  from action_controller/metal/instrumentation.rb:32:in `process_action'
  from action_controller/metal/params_wrapper.rb:256:in `process_action'
  from active_record/railties/controller_runtime.rb:24:in `process_action'
  from scout_apm/instruments/action_controller_rails_3_rails4.rb:86:in `process_action'
  from abstract_controller/base.rb:134:in `process'
  from action_view/rendering.rb:32:in `process'
  from action_controller/metal.rb:191:in `dispatch'
  from action_controller/metal.rb:252:in `dispatch'
  from action_dispatch/routing/route_set.rb:52:in `dispatch'
  from action_dispatch/routing/route_set.rb:34:in `serve'
  from action_dispatch/journey/router.rb:52:in `block in serve'
  from action_dispatch/journey/router.rb:35:in `each'
  from action_dispatch/journey/router.rb:35:in `serve'
  from action_dispatch/routing/route_set.rb:840:in `call'
  from scout_apm/instruments/rails_router.rb:29:in `call_with_scout_instruments'
  from scout_apm/middleware.rb:17:in `call'
  from warden/manager.rb:36:in `block in call'
  from warden/manager.rb:34:in `catch'
  from warden/manager.rb:34:in `call'
  from rack/tempfile_reaper.rb:15:in `call'
  from rack/etag.rb:25:in `call'
  from rack/conditional_get.rb:25:in `call'
  from rack/head.rb:12:in `call'
  from app/middlewares/sanitize_null_bytes.rb:14:in `call'
  from action_dispatch/http/content_security_policy.rb:18:in `call'
  from rack/session/abstract/id.rb:232:in `context'
  from rack/session/abstract/id.rb:226:in `call'
  from action_dispatch/middleware/cookies.rb:670:in `call'
  from action_dispatch/middleware/callbacks.rb:28:in `block in call'
  from active_support/callbacks.rb:98:in `run_callbacks'
  from action_dispatch/middleware/callbacks.rb:26:in `call'
  from action_dispatch/middleware/debug_exceptions.rb:61:in `call'
  from ddtrace/contrib/rails/middlewares.rb:16:in `call'
  from action_dispatch/middleware/show_exceptions.rb:33:in `call'
  from rails/rack/logger.rb:38:in `call_app'
  from rails/rack/logger.rb:26:in `block in call'
  from active_support/tagged_logging.rb:71:in `block in tagged'
  from active_support/tagged_logging.rb:28:in `tagged'
  from active_support/tagged_logging.rb:71:in `tagged'
  from rails/rack/logger.rb:26:in `call'
  from action_dispatch/middleware/remote_ip.rb:81:in `call'
  from request_store/middleware.rb:19:in `call'
  from action_dispatch/middleware/request_id.rb:27:in `call'
  from rack/method_override.rb:22:in `call'
  from rack/runtime.rb:22:in `call'
  from active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
  from action_dispatch/middleware/executor.rb:14:in `call'
  from action_dispatch/middleware/static.rb:127:in `call'
  from rack/sendfile.rb:111:in `call'
  from rack/cors.rb:98:in `call'
  from raven/integrations/rack.rb:51:in `call'
  from ddtrace/contrib/rack/middlewares.rb:85:in `call'
  from scout_apm/instruments/middleware_summary.rb:58:in `call'
  from rails/engine.rb:524:in `call'
  from puma/configuration.rb:228:in `call'
  from puma/server.rb:664:in `handle_request'
  from puma/server.rb:467:in `process_client'
  from puma/server.rb:328:in `block in run'
  from puma/thread_pool.rb:135:in `block in spawn_thread'
```

**Comment by ioquatix** [comment]
This is happening because data dog is using legacy wrappers for methods rather than modern Ruby 2.x `prepend` style wrappers.

**Comment by ioquatix** [comment]
@w32-blaster what version of rails are you using? Do you mind commenting on the linked DataDog issue?

**Comment by w32-blaster** [comment]
Right now 5.2.3

**Comment by adamcrown** [comment]
I'm seeing this issue too. For now I've just worked around it by disabling `NetHttp` in the `scout_apm.yml` file.

```yaml
development:
  disabled_instruments:
    - NetHttp
```

**Comment by lloeki** [comment]
I think this should be closed as long as the the following combination is used, as that's when both moved to `prepend`:

- `scout_apm` >= 2.5.2
- `ddtrace` >= 0.27

**Comment by jrothrock** [comment]
Unfortunately this is still occurring with `scout_apm` 2.6.10 and `ddtrace` 0.42:
```
SystemStackError: stack level too deep
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/configuration/settings.rb:15:in `block (2 levels) in <class:Settings>'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/configuration/option.rb:60:in `instance_eval'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/configuration/option.rb:60:in `context_eval'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/configuration/option.rb:27:in `get'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/configuration/options.rb:72:in `get_option'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/configuration/options.rb:43:in `block in default_helpers'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/configuration/settings.rb:30:in `[]'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:116:in `datadog_pin'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:38:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/ddtrace-0.42.0/lib/ddtrace/contrib/http/instrumentation.rb:42:in `request'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:34:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/tracer.rb:44:in `instrument'
      /opt/apps/source/conduit/shared/bundle/ruby/2.5.0/gems/scout_apm-2.6.10/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
```

**Comment by dlanderson** [comment]
There are still some places where we use `alias_method`. We'll be moving those to use `prepend` soon.

**Comment by sposin** [comment]
Hi, we have this issue as well. Error is 
```fatal - machine stack overflow in critical region:```

ddtrace 0.45.0
scout_apm 4.0.4

I tried, but it did not make a difference.
```
development:
  disabled_instruments:
    - NetHttp
```

ruby 3.0.0
rails 6.0.3.5

**Comment by dlanderson** [comment]
@sposin Typically this issue would raise a `SystemStackError` exception - would you be able to provide a backtrace or more detailed output from the error so we can see which interaction might be causing your issue?

**Comment by sposin** [comment]
> @sposin Typically this issue would raise a `SystemStackError` exception - would you be able to provide a backtrace or more detailed output from the error so we can see which interaction might be causing your issue?

I wish I could be more helpful, but the only error I receive after enabling scout is
```fatal - machine stack overflow in critical region:```

**Comment by dlanderson** [comment]
@sposin Odd, there's usually more following the `fatal - machine stack overflow in critical region:` line which would give us an idea of where the overflow happened. We'll try to repro with your versions listed. If it's not giving us any more info, I'm not sure how we can determine the cause.

**Comment by sposin** [comment]
> @sposin Odd, there's usually more following the `fatal - machine stack overflow in critical region:` line which would give us an idea of where the overflow happened. We'll try to repro with your versions listed. If it's not giving us any more info, I'm not sure how we can determine the cause.

I removed `ddtrace` from the project and still have the fatal errors, so the issue is not related to that gem as was suggested. I am happy to help debug,  but perhaps that should be on a different issue?

**Comment by dlanderson** [comment]
> I removed `ddtrace` from the project and still have the fatal errors, so the issue is not related to that gem as was suggested. I am happy to help debug, but perhaps that should be on a different issue?

Yes if you don't mind let's start a new issue.

**Comment by samsonjs** [comment]
We ran into a similar issue with Sentry: https://github.com/getsentry/sentry-ruby/issues/1496

Stack trace for reference:

```
[07/06/21 18:04:11 +0000 worker-f1cd15d4f808 (18)] INFO : Exception sending request to server: 
stack level too deep
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/config.rb:226:in `each'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/config.rb:226:in `detect'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/config.rb:226:in `overlay_for_key'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/config.rb:234:in `value'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:43:in `request_scout_description'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
	/usr/local/bundle/gems/sentry-ruby-core-4.6.0/lib/sentry/net/http.rb:38:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:34:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:44:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
	/usr/local/bundle/gems/sentry-ruby-core-4.6.0/lib/sentry/net/http.rb:38:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:34:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:44:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
	/usr/local/bundle/gems/sentry-ruby-core-4.6.0/lib/sentry/net/http.rb:38:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:34:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:44:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:29:in `request_with_scout_instruments'
	/usr/local/bundle/gems/sentry-ruby-core-4.6.0/lib/sentry/net/http.rb:38:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/instruments/net_http.rb:30:in `block in request_with_scout_instruments'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:34:in `instrument'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/tracer.rb:44:in `instrument'
[... thousands of lines...]
	/usr/local/bundle/gems/sentry-ruby-core-4.6.0/lib/sentry/net/http.rb:38:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:85:in `block in post'
	/usr/local/lib/ruby/3.0.0/net/http.rb:960:in `start'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:92:in `request'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:80:in `post'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:158:in `block in post_payload'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:156:in `each'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:156:in `post_payload'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporter.rb:40:in `report'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporting.rb:93:in `deliver_period'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporting.rb:52:in `block in report_to_server'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/layaway.rb:103:in `with_claim'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporting.rb:45:in `report_to_server'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/reporting.rb:36:in `process_metrics'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/periodic_work.rb:13:in `run'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/agent.rb:178:in `block (2 levels) in start_background_worker'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/background_worker.rb:61:in `block in start'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/background_worker.rb:44:in `loop'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/background_worker.rb:44:in `start'
	/usr/local/bundle/gems/scout_apm-4.0.4/lib/scout_apm/agent.rb:177:in `block in start_background_worker'
[07/06/21 18:22:11 +0000 worker-f1cd15d4f808 (18)] WARN : Error on checkin to https://checkin.scoutapp.com/apps/checkin.scout?key=redacted&name=redacted: :connection_failed
[2021-07-06 18:22:25.694] ERROR -- stack level too deep: nil
```

**Comment by rusucosmin** [comment]
+1 - we eneded up removing scout_apm from our Gemfile

**Comment by dlanderson** [comment]
@rusucosmin Apologies - If you list `scout_apm` above the datadog gem in your Gemfile it should work. Until we can move to `prepend` for our instruments, this is the recommended workaround for a `stack level too deep` instrumentation conflict
